Library/Homebrew/formula.rb

```diff
@@ -1479,9 +1479,11 @@ def std_configure_args
   end
 
   # Standard parameters for cargo builds.
-  sig { returns(T::Array[T.any(String, Pathname)]) }
-  def std_cargo_args
-    ["--locked", "--root", prefix, "--path", "."]
+  sig { params(locked: T::Boolean, root: String, path: String).returns(T::Array[T.any(String, Pathname)]) }
+  def std_cargo_args(locked: true, root: prefix, path: ".")
+    args = ["--root", root, "--path", path]
+    args.prepend "--locked" if locked
+    args
```

Comment:
Let's make this a bit more consistent with how we usually construct arrays of flags in formulae:
```suggestion
    args = []
    args << "--locked" if locked
    args += ["--root", root, "--path", path]

    args
```
Not sure we really need to change `--path`, since that can actually be fixed with a `cd` to the appropriate directory, as some formulae already do now. But I guess it doesn't hurt.

Reply:
Oddly enough, that was originally how I implemented it, so that works for me.

---

> Not sure we really need to change `--path`, since that can actually be fixed with a `cd` to the appropriate directory, as some formulae already do now. But I guess it doesn't hurt.

I'll have to do some testing but my thinking was that there may be a functional difference between `cd` and using `cargo install --path` if a project is using workspaces (e.g., `gping`).

Reply:
Can we add an audit that prevents the use of `locked: false` in Homebrew/core? I don't think we ever want that there.

Reply:
@carlocab @samford I'd be tempted to not make it configurable at all unless we need it.

Reply:
The only formula not using `--locked` is `git-series`, as it was one of the exceptions in the late 2019 campaign to add `--locked` to Rust formulae. We weren't able to add `--locked` because it would break the build due to a dependency needing to be updated. Upstream updated the dependencies in a later commit to fix the issue but the changes haven't been included in a release. The last commit was in 2019, so this doesn't seem likely to change anytime soon.

I can't imagine a situation where we would remove `--locked` from a formula that's currently using it, so I'm good with making it non-configurable. We can simply leave `git-series` as it is and update it to use `std_cargo_args` in the future if they ever cut a new release.

If we ever encounter a situation where it makes sense to omit `--locked` in the future (doubtful), we can always revisit this.

Reply:
> We can simply leave `git-series` as it is and update it to use `std_cargo_args` in the future if they ever cut a new release.

👍🏻 sounds good.